dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java

```diff
@@ -130,6 +137,7 @@ public int deleteSoftDeletedProgramInstances()
             "delete from programmessage_emailaddresses where programmessageemailaddressid in " + pmSelect,
             "delete from programmessage_phonenumbers where programmessagephonenumberid in " + pmSelect,
             "delete from programmessage where programinstanceid in " + piSelect,
+            "delete from programmessage where programstageinstanceid in " + psiSelect,
```

Comment:
should it be
```suggestion
            "delete from programmessage where programinstanceid in " + piSelect,
```
??
or maybe it's not clear for me

Reply:
while deleting enrollment we also have to delete its associated event's program messages